src/main/java/org/mqttbee/mqtt/handler/subscribe/MqttSubscriptionHandler.java

```diff
@@ -245,13 +282,21 @@ private void handleComplete(@NotNull final ChannelHandlerContext ctx, final int
         }
     }
 
-    private static boolean allErrorCodes(@NotNull final ImmutableList<? extends Mqtt5ReasonCode> reasonCodes) {
+    private static int validateReasonCodes(
+            final int count, @NotNull final ImmutableList<? extends Mqtt5ReasonCode> reasonCodes) {
+
+        if (reasonCodes == Mqtt3UnsubAckView.REASON_CODES_ALL_SUCCESS) {
+            return REASON_CODES_VALIDATED;
+        }
+        if (reasonCodes.size() != count) {
+            return REASON_CODES_COUNT_DOES_NOT_MATCH;
+        }
         for (final Mqtt5ReasonCode reasonCode : reasonCodes) {
             if (!reasonCode.isError()) {
-                return false;
+                return REASON_CODES_VALIDATED;
```

Comment:
The name REASON_CODES_VALIDATED is misleading. Here it only means there is at least one reason code that's not an error.